src/index.js

```diff
@@ -18,7 +18,8 @@ export const parse = (remote) => {
     remote.share = share
   } else if (type === 'smb') {
     remote.type = 'smb'
-    const [auth, smb] = rest.split('@')
+    const smb = rest.slice(rest.lastIndexOf('@'))
+    const auth = rest.slice(0, rest.lastIndexOf('@'))
```

Comment:
It's ok for this time, but in the future, use a variable to store the index and avoid unnecessary calls.